### PR TITLE
XML with valid licenses, regardless of input

### PR DESCRIPTION
**Pull request by jkowalleck**
the current implementation has the change to create invalid XML:
when a dependency uses a license that is unknown to the [XML schema].

licenses are values of an enumeration
described here: https://cyclonedx.org/schema/spdx
according to [XML schema].

this patch does the following:
* add tests for licenses that are invalid.
* ship a list of known licenses.
* licenses in the XML will be written as `<id>` or `<name>` - just as described in the [XML schema].
* add a script to download latest known licenses.

[XML schema]: https://cyclonedx.org/schema/spdx

### CODE REVIEW

**Comment by jkowalleck** [comment]
on my way to this solution i thought about using https://packagist.org/packages/composer/spdx-licenses
but then i realized, that the list of valid licenses in the  https://cyclonedx.org/schema/spdx
might differ from `composer/spdx-licenses`.

so i added an own implementation and shipment, to be compliant to the XSD

**Comment by stevespringett** [comment]
Thanks for working on this PR. Please use the same license list and file name found here:
https://github.com/CycloneDX/cyclonedx-node-module/blob/master/spdx-licenses.json

This is auto-generated at the same time as https://cyclonedx.org/schema/spdx is updated. I will updated the auto-generation to also produce it for PHP as well.

**Comment by jkowalleck** [comment]
thanks for the hint, @stevespringett 

> I will update the auto-generation to also produce it for PHP as well.
could you explain what you are planning to do?
will there be a job that pushes a file  - just as https://github.com/CycloneDX/cyclonedx-node-module/blob/master/spdx-licenses.json - to this very repository?
If so, just drop me a hint when this json was merged to master the first time, so i will rebase/rework my PR to use the provided json and drop the `CycloneDX\Spdx\XmlLicensesUpdater`

**Comment by stevespringett** [comment]
This is what I currently run:
https://github.com/CycloneDX/specification/blob/master/tools/src/main/java/org/cyclonedx/tools/SpdxXsdGenerator.java

I currently run it manually. It generates the spdx.xsd used by the core specification. But it also generates what I need for the Java Core library (reference implementation) and the node module (which I pointed out earlier). I'll simply add the PHP to the list of locations where it needs updating.

Eventually, I want to turn this into a GitHub action that can be triggered and the resulting files will automatically be committed to their respective repositories.

**Comment by jkowalleck** [comment]
got it.
copied over the https://github.com/CycloneDX/cyclonedx-node-module/blob/master/spdx-licenses.json
maually for now.

removed XML SPDX Updater.

File: res/xml-spdx-licenses.json [hunk 1]
```diff
@@ -0,0 +1,485 @@
+[
```
Comment (jkowalleck) [inline res/xml-spdx-licenses.json:1]: copy from https://raw.githubusercontent.com/CycloneDX/cyclonedx-node-module/master/spdx-licenses.json

**Pull request merged**